### DISCUSSION TITLE
Platform-specific CLI configuration for command without duplication

### DISCUSSION THREAD

**Discussion by smackesey** [tox-dev/tox#2371] [Q&A] [answered]
I am trying to define a tox.ini that defines platform-dependent config for a single CLI argument in a single command. My current solution uses an environment variable:

```
commands:
  pylint -j {env:PYLINT_JOBS:0} ...
```

I would like to get rid of the environment variable and infer the correct setting based on platform. If on macOS, `-j 1` should be passed, on all other platforms `-j 0` should be passed. I know I can achieve this with:

```
[tox]
envlist = pylint-{macos,other}

[testenv]
commands =
  macos: pylint -j 1 ...
  other: pylint -j 0 ...
```

I don't like this solution because:

- It duplicates the entire rest of the command
- It requires the tox invocation to explicitly specify the platform, even though this information can be inferred from `sys.platform`

Is there any solution that avoids both of the above? I'm hoping for something that basically works like this:

```
[testenv]
commands =
  pylint -j {sys.platform == 'darwin' ? '1' : '0'} ...
```

**Comment by jugmac00** [comment, marked as answer]
I like your second approach the most - this is the one I see quite often in a tox configuration.

If you want to avoid the duplication, you can always create a bash or a Python script as a wrapper around your pylint command.

**Comment by Czaki** [comment]
It is also possible to workaround this using `toxfile.py`. The toxfile is a local tox plugin. In my example, I need to have the name of the used platform (to avoid collision of report files). So I use the following code:

```python
from __future__ import annotations

import sys
from typing import TYPE_CHECKING

from tox.plugin import impl

if TYPE_CHECKING:
    from tox.config.sets import EnvConfigSet
    from tox.session.state import State

@impl
def tox_add_env_config(env_conf: EnvConfigSet, state: State) -> None:
    env_conf.add_constant(["sys_platform"], "string representing current OS", sys.platform)
    print("tox_add_env_config", env_conf, state)
```
And then could use `sys_platform` as the substitution in the `commands` definition.
